api/src/main/java/org/openmrs/module/appointments/conflicts/impl/PatientDoubleBookingConflict.java

```diff
@@ -0,0 +1,59 @@
+package org.openmrs.module.appointments.conflicts.impl;
+
+import org.openmrs.module.appointments.conflicts.AppointmentConflictType;
+import org.openmrs.module.appointments.dao.AppointmentDao;
+import org.openmrs.module.appointments.model.Appointment;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.openmrs.module.appointments.constants.AppointmentConflictTypeEnum.PATIENT_DOUBLE_BOOKING;
+
+public class PatientDoubleBookingConflict implements AppointmentConflictType {
+
+    private AppointmentDao appointmentDao;
+
+    public void setAppointmentDao(AppointmentDao appointmentDao) {
+        this.appointmentDao = appointmentDao;
+    }
+
+    @Override
+    public String getType() {
+        return PATIENT_DOUBLE_BOOKING.name();
+    }
+
+    @Override
+    public List<Appointment> getAppointmentConflicts(List<Appointment> appointments) {
+        List<Appointment> conflictingAppointments = new ArrayList<>();
+        List<Appointment> patientAppointments = getPatientAppointments(appointments.get(0).getPatient().getPatientId());
```

Comment:
possiblility of null pointer exception  `appointments.get(0)`

Reply:
we are calling this method only when the list is non-empty.

Reply:
That is acceptable only for private methods since it is public, null case should be hanlde

Reply:
Added null check done